lib/dataset/groupby.cpp

```diff
@@ -166,27 +169,45 @@ template <class T> T GroupBy<T>::concat(const Dim reductionDim) const {
 
 /// Reduce each group using `sum` and return combined data.
 template <class T> T GroupBy<T>::sum(const Dim reductionDim) const {
-  return reduce(sum_impl, reductionDim, FillValue::ZeroNotBool);
+  return reduce(sum_into, reductionDim, FillValue::ZeroNotBool);
 }
 
 /// Reduce each group using `all` and return combined data.
 template <class T> T GroupBy<T>::all(const Dim reductionDim) const {
-  return reduce(all_impl, reductionDim, FillValue::True);
+  return reduce(
+      [](Variable &out, const Variable &var) {
+        accumulate_in_place(out, var, core::element::logical_and_equals,
+                            "groupby.all");
+      },
+      reductionDim, FillValue::True);
 }
 
 /// Reduce each group using `any` and return combined data.
 template <class T> T GroupBy<T>::any(const Dim reductionDim) const {
-  return reduce(any_impl, reductionDim, FillValue::False);
+  return reduce(
+      [](Variable &out, const Variable &var) {
+        accumulate_in_place(out, var, core::element::logical_or_equals,
+                            "groupby.any");
+      },
```

Comment:
It occurs to me now that we can (and imho should) avoid calling `accumulate_in_place` here separately. It is the same calls as in `variable/reduction.cpp` (except for the string names), which means we have duplicated code generation, leading to longer compile times and binary sizes. And note that we are in a templated class here, so this will get instantiated twice (for a total of 3, instead of 1).

Reply:
Except that `reduce_impl` was also a template and got instantiated for every kernel. All that has changed is that it is now done in the dataset module instead of variable.

Reply:
Not sure I understand your comment? Of course we need one instantiation per kernel? My point is that we should not make 3 instantiations for the same kernel.

Reply:
But all 3 are identical. Does the compiler go through each one from scratch and only match them up at the end or does it quickly see that it already has an instance?

Reply:
They are in different translation units. So it has to compile them (at least twice), and then you can *hope* that LTO removes them?

Reply:
Actually no, LTO won't help: We have an instantiation in `libscipp-variable.so` and one on `libscipp-dataset.so`, I doubt it can remove it from the latter and point to the former?

Reply:
Where do we instantiate this in variable?

Reply:
Oh, in `reduce_idempotent`.

But on the other hand, we are now not exporting the `all_impl` etc functions from `variable`. So this reduces binary size.

Reply:
Yes that reduced the size, but why trade it for making another binary larger? I am not sure what your concern is. Why not add a simple wrapper function?